ath79: add support for tp-link tl-wr1045nd v2

RasimKalimullin opened the pull request:
Initial support tp-link tl-wr1045nd v2, clone of tl-wr1043nd v3. 
Device url: https://www.tp-link.com/ru/home-networking/wifi-router/tl-wr1045nd/#specifications
Discussion: http://4pda.ru/forum/index.php?showtopic=664252&st=460


As I understand it, the device is only for the russian market.

adrianschmutzler commented:
Hi, your PR has formal issues, please have a look at https://openwrt.org/submitting-patches

Despite, you are mixing ath79 and ar71xx in your commit. Those are separate targets, which each theoretically could provide support for the device.
However, ar71xx is EOL and will only be supported up to the most recent stable branch, so no new devices will be accepted. So, please focus on adding device support to ath79 (as you partially did in your commit), and remove all files from ar71xx.

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -273,6 +273,10 @@ ath79_setup_interfaces()
+	tplink,tl-wr1045nd-v2)
+		ucidef_add_switch "switch0" \
+			"0@eth1" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1" "5:wan" "6@eth0"
+		;;
```
This can be merged with the case above.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_tplink_tl-wr1045nd-v2.dts:
```diff
@@ -0,0 +1,12 @@
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
```
You don't need those two includes, they are already in the dtsi you link below.

adrianschmutzler commented on target/linux/ath79/image/generic-tp-link.mk:
```diff
@@ -440,6 +440,17 @@ define Device/tplink_tl-wr1043n-v5
+  DEVICE_VARIANT := v2
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ledtrig-usbport
+  TPLINK_HWID := 0x10450002
+  SUPPORTED_DEVICES += tl-wr1045nd-v2
```
This line can be dropped, it's only required for update from ar71xx.

RasimKalimullin commented:
ok,thx, work will continue

adrianschmutzler commented:
Thanks for addressing my comments.

Note that you will have to squash all your commit into one and then push that one to your remote branch with "git push -f".

You still lack some of the formal requirements, too, like commit description and target prefix to the commit title "ath79: ".

RasimKalimullin commented:
Hi! I hope I did it.

adrianschmutzler commented:
Coming closer.

You still need to:
- remove your manual "Author" and "Date" lines, as well as the "Changes to be committed".
- adjust the "From:" line in the patch view to match your Signed-off. This can be done easily by "git commit --amend --reset-author"
- add device specifications to your commit message (e.g. https://github.com/openwrt/openwrt/commit/bba6646b5cba591946b3fd0caaad79cb6ec78043)
- add TFTP flashing instructions (if possible)
- (optional) Capitalize device name in commit message: "ath79: add support for TP-Link TL-WR1045ND v2"

adrianschmutzler commented:
BTW: Is there any particular reason why you call this "initial" support in the PR title?
This imply that support is only partial, and features are supposed to be added later.
If that's the case, please specify what's "missing" at the moment. If not, maybe change "Initial support" to "Add support for" in the PR title ...

RasimKalimullin commented:
At the moment I do not see a problem. Hardware is the same with 1043nd v3, everything works correctly.

adrianschmutzler commented:
> At the moment I do not see a problem. Hardware is the same with 1043nd v3, everything works correctly.

That's what I suspected ;-)

RasimKalimullin committed:
ath79: add support for TP-Link TL-WR1045ND v2

Specifications:
 - QCA9558 Soc
 - 720/800 (CPU/DDR)
 - 1x 10/100/1000 Mbps WAN Ethernet
 - 4x 10/100/1000 Mbps LAN Ethernet
 - 64 MB RAM (DDR2)
 - 8 MB FLASH
 - QCA9558 2.4 GHz 802.11bgn
 - 1x USB 2.0

Flash instruction
WebUI:
Download *-factory.bin image, rename to tp_firmware.bin and upload
it via the firmwary upgrade function of the stock firmware WebUI.

Tftp:
Rename OpenWRT or original firmware to wr1045v2_tp_recovery.bin and
Change your computer ip in 192.168.0.66 and subnet mask in 255.255.255.0.
Router will obtain IP 192.168.0.86 for a few seconds while loading,
when reset button pressed at power On.

Signed-off-by: Rasim Kalimullin <neutrino.vm@gmail.com>

RasimKalimullin commented:
> Coming closer.
> 
> You still need to:
...

Dear Adrian! 
What else is missing? 
Thank you for your patience.

adrianschmutzler commented:
I do not see anything obvious at the moment. However, I cannot merge this, so you will have to wait for another review anyway.

adrianschmutzler commented:
I've rebased the base-files change and will merge this to master in the next days.
Thanks for your contribution.
